rust/connlib/tunnel/src/peer.rs

```diff
@@ -19,20 +18,17 @@ type ExpiryingResource = (ResourceId, Option<DateTime<Utc>>);
 // is 30 seconds. See resolvconf(5) timeout.
 const IDS_EXPIRE: std::time::Duration = std::time::Duration::from_secs(60);
 
-pub struct Peer<TId, TTransform, TResource> {
-    // TODO: we should refactor this
-    // in the gateway-side this means that we are explicit about ()
-    // maybe duping the Peer struct is the way to go
-    pub allowed_ips: IpNetworkTable<TResource>,
-    pub conn_id: TId,
-    pub transform: TTransform,
+/// The state of one gateway on a client.
+pub(crate) struct GatewayOnClient {
```

Comment:
```suggestion
pub(crate) struct GatewayPeer {
```

perhaps?

I just don't like `GatewayOnClient` too much

Reply:
Maybe let's move each of these to different modules?

Reply:
> Maybe let's move each of these to different modules?

I am happy to do so in a follow-up PR. I took care to make the diff as minimal as possible so that it is easy to see that we are just splitting up this struct but the functionality of transforming the packets is actually untouched.

Reply:
> I just don't like `GatewayOnClient` too much

I am open to other names. Initially I just had this named `Gateway` because it represents the state of a `Gateway` that a client is connected to. But it confused the shit out of me haha

So when I started to write some docs, it dawned on me that it is the "gateway state on a client" so I went the pragmatic way and named it that.

Reply:
If you don't mind, I'd merge it like this and we can iterate on better names in follow-up PRs!